src/Cache/CacheTrait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Cache;
+
+use Psr\Cache\CacheException;
+use Psr\Cache\CacheItemPoolInterface;
+
+trait CacheTrait
+{
+    private $localCache = [];
+
+    private function getOrSave(string $cacheKey, CacheItemPoolInterface $pool, callable $callback)
```

Comment:
naming proposal:

```php
<?php
private function cache(string $cacheKey, callable $getValue) {

}
```

With a parent class we can remove CacheItemPoolInterface from the method signature

Reply:
We don't need a parent class. The property could be declared in the trait as we already do elsewhere?

Reply:
also works yes, I just think that parent class > trait if we can :) (ability to use `instanceof parent` etc.)

Reply:
I think using inheritance here is... wrong. It's the kind of use case which traits are exactly designed to solve - horizontal code reuse.